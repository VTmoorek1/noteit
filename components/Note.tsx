import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Media from './Media';
import '../stylesheets/note.css';
import { RemoveNoteItem } from '../interfaces/noteinterface';

interface Props {
    title : string,
    mID : string,
    desc : string,
    file : Blob,
    remove : (remObj : RemoveNoteItem) => void
}

/**
 * Note is a container for a user (future enhancement), title, 
 * description, and media component. These will fill the current page
 * loaded
 */
export default class Note extends Component<Props> {

    constructor(props : Props) {
        super(props);
    }

    render() {

        const {mID,title,file,desc,remove} = this.props;
        const remNote = {id : mID, name : title};

        return <div className="note bg-dark">
            <div id="topDiv">
                <img src="https://s3.amazonaws.com/cdn-origin-etr.akc.org/wp-content/uploads/2017/11/13000724/Belgian-Malinois-On-White-01.jpg" className="userImg" />
                <div id="titleDiv">
                    <h3 id="title">{title}</h3>
                </div>
                <div id="topBtnDiv">
                    <button onClick={() => remove(remNote)} 
                    className="btn btn-danger circleButtons" id="cancelNoteBtn" type="button"><i className="fa fa-times"></i></button>
                </div>
            </div>
            <div id="midDiv">
                <Media file={file} mediaID={mID} />
            </div>
            <div id="bottomDiv">
                <h6>{desc}</h6>
            </div>
        </div>;

    }

} 
