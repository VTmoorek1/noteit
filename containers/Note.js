import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Media from './Media';

/**
 * Note is a container for a user (future enhancement), title, 
 * description, and media component. These will fill the current page
 * loaded
 */
export default class Note extends Component {

    constructor(props) {
        super(props);
    }

    render() {

        const remNote = {id :this.props.mID, name : this.props.title};

        return <div className="note bg-dark">
            <div id="topDiv">
                <img src="https://s3.amazonaws.com/cdn-origin-etr.akc.org/wp-content/uploads/2017/11/13000724/Belgian-Malinois-On-White-01.jpg" id="userImg" />
                <div id="titleDiv">
                    <h3 id="title">{this.props.title}</h3>
                </div>
                <div id="topBtnDiv">
                    <button onClick={() => this.props.remove(remNote)} 
                    className="btn btn-danger circleButtons" id="cancelNoteBtn" type="button"><i className="fa fa-times"></i></button>
                </div>
            </div>
            <div id="midDiv">
                <Media file={this.props.file} mediaID={this.props.mID} />
            </div>
            <div id="bottomDiv">
                <h6>{this.props.desc}</h6>
            </div>
        </div>;

    }

} 

Note.propTypes = {
    title : PropTypes.string.isRequired,
    mID : PropTypes.string.isRequired,
    desc : PropTypes.string.isRequired,
    file : PropTypes.object.isRequired,
    remove : PropTypes.func
}