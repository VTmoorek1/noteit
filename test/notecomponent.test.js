import React from 'react';
import {shallow,mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Note from '../components/Note';

configure({adapter : new Adapter()});

const noteData = {
    file : new Blob(['blobber'], { type: 'gif' }),
    title : 'Note Title', 
    desc: 'This is the description',
    id: '12345678912345678901234'
};

it('should render a note with title, user, media and description', () => {

    const wrapper = mount(<Note mID={noteData.id} file={noteData.file} title={noteData.title}  desc={noteData.desc}/>);

    expect(wrapper).toMatchSnapshot();
});

it('should have the same props that were passed in', () => {

    const wrapper = mount(<Note mID={noteData.id} file={noteData.file} title={noteData.title}  desc={noteData.desc}/>);

    expect(wrapper.prop('desc')).toEqual(noteData.desc);
    expect(wrapper.prop('mID')).toEqual(noteData.id);
    expect(wrapper.prop('title')).toEqual(noteData.title);
    expect(wrapper.prop('file')).toEqual(noteData.file);
});

