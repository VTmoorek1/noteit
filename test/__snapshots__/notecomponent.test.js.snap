// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a note with title, user, media and description 1`] = `
<Note
  desc="This is the description"
  file={Blob {}}
  mID="12345678912345678901234"
  title="Note Title"
>
  <div
    className="note bg-dark"
  >
    <div
      id="topDiv"
    >
      <img
        className="userImg"
        src="https://s3.amazonaws.com/cdn-origin-etr.akc.org/wp-content/uploads/2017/11/13000724/Belgian-Malinois-On-White-01.jpg"
      />
      <div
        id="titleDiv"
      >
        <h3
          id="title"
        >
          Note Title
        </h3>
      </div>
      <div
        id="topBtnDiv"
      >
        <button
          className="btn btn-danger circleButtons"
          id="cancelNoteBtn"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fa fa-times"
          />
        </button>
      </div>
    </div>
    <div
      id="midDiv"
    >
      <Media
        file={Blob {}}
        mediaID="12345678912345678901234"
      >
        <div
          id="mediaDiv"
        />
      </Media>
    </div>
    <div
      id="bottomDiv"
    >
      <h6>
        This is the description
      </h6>
    </div>
  </div>
</Note>
`;
